### PR TITLE
java core: Change all constructor version args to int.

### PR DESCRIPTION
**Pull request by tstein**

### CODE REVIEW

File: java/core/src/main/java/com/microsoft/bond/protocol/CompactBinaryReader.java [hunk 1]
```diff
@@ -28,7 +28,7 @@ public CompactBinaryReader(InputStream inputStream, short version) {
         }
 
         this.reader = new BinaryStreamReader(inputStream);
-        this.version = version;
+        this.version = (short) version;
```
Comment (chwarr) [inline java/core/src/main/java/com/microsoft/bond/protocol/CompactBinaryReader.java]: Should there be a range check before the cast?

Comment (tstein) [inline java/core/src/main/java/com/microsoft/bond/protocol/CompactBinaryReader.java]: We'll throw on line 27 if version isn't 1 or 2.

Comment (chwarr) [inline java/core/src/main/java/com/microsoft/bond/protocol/CompactBinaryReader.java]: Ahh... The joys of diff folding... Thanks.

**Comment by tstein** [comment]
Removed casts in tests and examples.

**Pull request merged**